clang/lib/CodeGen/CGOpenMPRuntime.cpp

```diff
@@ -2727,13 +2730,15 @@ CGOpenMPRuntime::emitSeverityClause(OpenMPSeverityClauseKind Severity) {
 llvm::Value *
 CGOpenMPRuntime::emitSeverityClause(const OMPSeverityClause *SeverityClause) {
   return emitSeverityClause(SeverityClause ? SeverityClause->getSeverityKind()
-                                           : OMPC_SEVERITY_unknown);
+                                           : OMPC_SEVERITY_unknown,
+                            SeverityClause->getBeginLoc());
```

Comment:
You perform a nullptr check the first time dereferencing `SeverityClause` but skip the check the second time and unconditionally dereference it. One of these is a bug.